### DISCUSSION TITLE
Chinese characters in itwin schema are encoded into English format strings. Is there a way to decode them back to the original Chinese characters?

### DISCUSSION THREAD

**Discussion by shenshengyi** [iTwin/itwinjs-core#3699] [Q&A] [answered]
Hi，
I found that the itemtype I edited in the microstation contains Chinese characters. After converting to itwin, the Chinese names are all encoded, and I cannot use the original Chinese characters to query.
For example: <BaseClass>__x8BBE____x5907____x53C2____x6570__:__x6807____x7B7E__ElementAspect</BaseClass>
Is there a way for me to decode it and get the original Chinese characters? thanks.

**Comment by pmconne** [comment, marked as answer]
You can use the [ECName API](https://www.itwinjs.org/reference/ecschema-metadata/metadata/ecname/) to convert between the encoded name and the original Chinese name.

**Reply by shenshengyi** [reply]
solved, thanks.

**Reply by ColinKerr** [reply]
@shenshengyi the original characters should have been saved in the display label for each schema element.  These can be accessed via an ECSql query:

**ECSql**
```sql
SELECT Name, DisplayLabel FROM ECDbMeta.ECClassDef
```
This will select the name and display label for every class, you can do the same for schemas and properties (ECSchemaDef and ECPropertyDef respectively)
